pkg/ingester/index/multi.go

```diff
@@ -0,0 +1,123 @@
+package index
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/grafana/loki/pkg/querier/astmapper"
+	"github.com/grafana/loki/pkg/storage/config"
+)
+
+type periodIndex struct {
+	time.Time
+	idx int // address of the index to use
+}
+
+type Multi struct {
+	periods []periodIndex
+	indices []Interface
+}
+
+func NewMultiInvertedIndex(periods []config.PeriodConfig, indexShards uint32) (*Multi, error) {
+	var (
+		err error
+
+		ii          Interface // always stored in 0th index
+		bitPrefixed Interface // always stored in 1st index
+
+		periodIndices []periodIndex
```

Comment:
maybe you can assign `periodIndices := make([]periodIndex, len(periods))`

Reply:
Yeah, that would work, but the overhead here is basically zero (a 2-length slice that's instantiated once on startup)

Reply:
got it, I thought we could have a huge number of periods